File: backend/src/main/java/com/reservation/model/entity/Availability.java
package com.reservation.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "availabilities")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Availability {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teacher_id", nullable = false)
    private User teacher;
    
    @Column(nullable = false)
    private LocalDate availableDate;
    
    @Column(nullable = false)
    private LocalTime startTime;
    
    @Column(nullable = false)
    private LocalTime endTime;
}

File: frontend/src/types/Availability.ts
import { User } from './User';

export interface Availability {
  id: number;
  teacher: User;
  availableDate: string; // LocalDate as ISO string
  startTime: string; // LocalTime as string (HH:mm)
  endTime: string; // LocalTime as string (HH:mm)
}

export interface AvailabilityCreateRequest {
  teacherId: number;
  availableDate: string;
  startTime: string;
  endTime: string;
}
